### PR TITLE
Ensure that soft opt-ins are set for Android Feast in-app purchases

### PR DESCRIPTION
**Pull request by michaelbjacobson**
This updates the existing switch statement to ensure that Android Feast IAPs trigger the correct soft opt-ins.

### CODE REVIEW

File: typescript/src/soft-opt-ins/processSubscription.ts [hunk 1]
```diff
@@ -122,7 +122,9 @@ export async function processAcquisition(subscriptionRecord: ReadSubscription, i
         handleError(`Soft opt-in message send failed for subscriptionId: ${subscriptionId}. ${e}`)
     }
 
-    const isFeast = subscriptionRecord.platform === Platform.IosFeast;
+    const isFeast =
+        subscriptionRecord.platform === Platform.IosFeast ||
+        subscriptionRecord.platform === Platform.AndroidFeast;
```
Comment (MaelGNM) [inline typescript/src/soft-opt-ins/processSubscription.ts:127]: Technically this change is not needed for the soft opt-in, but is for the email. It's fine to leave it but you'll have a conflict with https://github.com/guardian/mobile-purchases/pull/1535

**Review by MaelGNM — APPROVED** [review]
👍

**Comment by tjmw** [comment]
Hi @michaelbjacobson I've merged #1535 this morning which has caused a conflict as @MaelGNM pointed out above. Let me know if I can help resolving! Also in #1535 I added a Jest test for this lambda for an `android-feast` subscription. There's a little bit of the test commented which I think could be useful to test the change you've made in this PR. If you search for `TODO: fix this when correct SOIs are sent` you should find it. Again let me know if I can help!

**Review by tjmw — APPROVED** [review]
Nice! 👍

**Pull request merged**